std/string.d

```diff
@@ -5798,36 +5798,67 @@ unittest
  * function, or any of the conversion functions.
  *
  * Params:
- *     s = the string to check
+ *     s = the string or random access range to check
  *     bAllowSep = accept separator characters or not
  *
  * Returns:
  *     $(D bool)
  */
-bool isNumeric(S)(S s, bool bAllowSep = false) if (isSomeString!S)
+bool isNumeric(S)(S s, bool bAllowSep = false) if (isSomeString!S ||
+    (isRandomAccessRange!S &&
+    hasSlicing!S &&
+    isSomeChar!(ElementType!S) &&
+    !isInfinite!S))
 {
     import std.algorithm.comparison : among;
+    import std.ascii : isASCII;
 
-    if (s.empty)
+    // ASCII only case insensitive comparison with two ranges
+    static bool asciiCmp(S1, S2)(S1 a, S2 b)
+    {
+        import std.algorithm.comparison : equal;
+        import std.algorithm.iteration : map;
+        import std.ascii : toLower;
+        import std.utf : byChar;
+        return a.map!toLower.equal(b.byChar.map!toLower);
```

Comment:
Still has .byChar here. I see you removed it from the calls, but then I think asciiCmp should explicitly take b as a string. Otherwise, someone is going to pass a non-string there.

The other way would be a generic b with .byChar at the call sites but not here.